components/outputStates.js

```diff
@@ -0,0 +1,23 @@
+// outportStates.js
```

Comment:
I think it makes sense to have an inputStates.js module, to manage the inputStates object, but it seems like the functionality in this module will be largely covered by in the states.js module.  Also, the output and error objects will be used differently by the Framework, so I think it would be simpler to just have separate output and error state objects directly in the VNI object -- no need for an array to hold them.  For these reasons I suggest dropping this module, and have its functionality covered by state.js and vni.js .